server/dbutils.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+	"github.com/joshprzybyszewski/cribbage/server/play"
+)
+
+var (
+	errInvalidUsername error = errors.New(`invalid username`)
+)
+
+func commitOrRollback(db persistence.DB, err *error) {
+	var err2 error
+	if *err != nil {
+		err2 = db.Rollback()
+	} else {
+		err2 = db.Commit()
+	}
+	if err2 != nil {
+		log.Printf("Could not commit/rollback after %+v: %+v\n", err, err2)
+	}
+}
+
+func handleAction(_ context.Context, db persistence.DB, action model.PlayerAction) error {
+	err := db.Start()
```

Comment:
Now that i'm seeing this, I think `Start()` should return a "transaction" struct... I'll think on it

Reply:
Have you thought on it?

Reply:
🤷